x/operator/types/expected_keepers.go

```diff
@@ -0,0 +1,99 @@
+package types
+
+import (
+	sdkmath "cosmossdk.io/math"
+	delegationtype "github.com/ExocoreNetwork/exocore/x/delegation/types"
+	tmprotocrypto "github.com/cometbft/cometbft/proto/tendermint/crypto"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type ExpectDelegationInterface interface {
+	DelegationStateByOperatorAssets(ctx sdk.Context, operatorAddr string, assetsFilter map[string]interface{}) (map[string]map[string]delegationtype.DelegationAmounts, error)
+	IterateDelegationState(ctx sdk.Context, f func(restakerID, assetID, operatorAddr string, state *delegationtype.DelegationAmounts) error) error
+	UpdateDelegationState(ctx sdk.Context, stakerID string, assetID string, delegationAmounts map[string]*delegationtype.DelegationAmounts) (err error)
+
+	UpdateStakerDelegationTotalAmount(ctx sdk.Context, stakerID string, assetID string, opAmount sdkmath.Int) error
+}
+
+type PriceChange struct {
+	OriginalPrice sdkmath.Int
+	NewPrice      sdkmath.Int
+	Decimal       uint8
+}
+type ExpectOracleInterface interface {
```

Comment:
Please rename to `OracleKeeper` and add comments describing the expected interface because I don't see these functions in the oracle PR or design.

Reply:
Thanks